Textfield/Textfield.jsx

```diff
@@ -13,7 +14,7 @@ export default class Textfield extends MaterialComponent {
     this.state = {
       showFloatingLabel: false
     };
-    this._mdcProps = ["fullwidth", "multiline"];
+    this._mdcProps = ["fullwidth", "multiline", "type"];
```

Comment:
Adding a prop in this index, puts it in a loop to be constructed as a class.
https://github.com/prateekbh/preact-material-components/blob/master/MaterialComponent.js#L35-L37

You can skip this and still it will as desired.

P.S.: only Boolean kind of props land here which end up becoming classes for components root elements.